pkg/whiteflag/test/whiteflag_test.go

```diff
@@ -0,0 +1,599 @@
+}
+
+func generateDotFileFromTangle(t *testing.T) string {
+
+	visitedBundles := make(map[string]tangle.CachedBundles)
+
+	bundleTxs := tangle.GetAllBundleTransactionHashes(100)
+	for _, hash := range bundleTxs {
+		cachedTx := tangle.GetCachedTransactionOrNil(hash)
+		if _, visited := visitedBundles[string(cachedTx.GetTransaction().GetBundleHash())]; visited == false {
+			bndls := tangle.GetBundlesOfTransactionOrNil(cachedTx.GetTransaction().GetTxHash(), false)
+			visitedBundles[string(cachedTx.GetTransaction().GetBundleHash())] = bndls
+		}
+		cachedTx.Release(true)
+	}
+
+	var milestones []string
+	var included []string
+	var ignored []string
+	var conflicting []string
+
+	dotFile := fmt.Sprintf("digraph %s\n{\n", t.Name())
+	for _, bndls := range visitedBundles {
+		//singleBundle := len(bndls) == 1
+		for _, bndl := range bndls {
+			shortBundle := shortened(bndl)
+
+			bundleHead := bndl.GetBundle().GetHead()
+			//if singleBundle {
+			trunk := bndl.GetBundle().GetTrunk(true)
+			if tangle.SolidEntryPointsContain(trunk) {
+				dotFile += fmt.Sprintf("\"%s\" -> \"%s\";\n", shortBundle, shortenedHash(trunk))
+			} else {
+				trunkBundles := tangle.GetBundlesOfTransactionOrNil(trunk, false)
+				dotFile += fmt.Sprintf("\"%s\" -> \"%s\";\n", shortBundle, shortened(trunkBundles[0]))
+				trunkBundles.Release()
+			}
+
+			branch := bndl.GetBundle().GetBranch(true)
+			if tangle.SolidEntryPointsContain(branch) {
+				dotFile += fmt.Sprintf("\"%s\" -> \"%s\";\n", shortBundle, shortenedHash(branch))
+			} else {
+				branchBundles := tangle.GetBundlesOfTransactionOrNil(branch, false)
+				dotFile += fmt.Sprintf("\"%s\" -> \"%s\";\n", shortBundle, shortened(branchBundles[0]))
+				branchBundles.Release()
+			}
+
+			if bndl.GetBundle().IsMilestone() {
+				milestones = append(milestones, shortBundle)
+			} else if bndl.GetBundle().IsConfirmed() {
+				if bndl.GetBundle().IsConflicting() {
+					conflicting = append(conflicting, shortBundle)
+				} else if bndl.GetBundle().IsValueSpam() {
+					ignored = append(ignored, shortBundle)
+				} else {
+					included = append(included, shortBundle)
+				}
+			}
+
+			bundleHead.Release()
+		}
+		bndls.Release()
+	}
+
+	for _, milestone := range milestones {
+		dotFile += fmt.Sprintf("\"%s\" [shape=Msquare];\n", milestone)
+	}
+	for _, conf := range conflicting {
+		dotFile += fmt.Sprintf("\"%s\" [style=filled,color=red];\n", conf)
+	}
+	for _, conf := range ignored {
+		dotFile += fmt.Sprintf("\"%s\" [style=filled,color=gray];\n", conf)
+	}
+	for _, conf := range included {
+		dotFile += fmt.Sprintf("\"%s\" [style=filled,color=green];\n", conf)
+	}
+
+	dotFile += "}\n"
+	return dotFile
+}
+
+/// Testcases
+
+func TestWhiteFlagWithMultipleConflicting(t *testing.T) {
+
+	// Fill up the balances
+	balances := make(map[string]uint64)
+	balances[string(generateAddress(t, seed1, 0))] = 1000
+
+	milestones := setupCoordinatorAndIssueInitialMilestones(t, balances, 3)
+
+	// Issue some transactions
```

Comment:
Could you please add some comments here? It's hard to read the testcase :)
For example what would be the final balance you would expect on which seed, or something like that.

Reply:
Do you mean as a comment of every bundle? At the end I check every seed

Reply:
OK, I see that without labels it looks difficult to read. My IDE was showing it to me like this:
<img width="1005" alt="Screenshot 2020-07-24 at 09 08 23" src="https://user-images.githubusercontent.com/5908067/88367968-4fbb9e00-cd8d-11ea-9d89-53bbcaa201ab.png">